### PR TITLE
Enhance regression test of dynamic buffer calculation

### PR DESCRIPTION
**Pull request by stephenxs**
### Description of PR

Summary:
Enhance regression test of dynamic buffer calculation
Signed-off-by: Stephen Sun <stephens@nvidia.com>

### Type of change

- [ ] Bug fix
- [ ] Testbed and Framework(new/improvement)
- [x] Test case(new/improvement)

### Back port request
- [ ] 201911

#### What is the motivation for this PR?
Enhance regression test of dynamic buffer calculation

#### How did you do it?

- Adjust test cases for 8-lane ports
  On some switches, ports with 8 lanes need special handling
  Port with/without 8 lanes will have different buffer profiles when they share the same speed
  - Number of lanes needs to be taken into account when generating the expected profiles
  - Make port_to_test a module-level fixture and introduce new global NUMBER_OF_LANES
    This is to hide the details regarding the number of lanes into the method to generate the expected profile
- Adjust buffer pool calculation according to private headroom
- Adjust expected error messages of test_exceeding_headroom
  Move some error messages to ignore list as they don't always happen in time during the test.
  Only error messages reported by `buffermgrd` are in the expected list

#### How did you verify/test it?
Run regression test.

### CODE REVIEW

File: tests/qos/test_buffer.py [hunk 1]
```diff
@@ -734,8 +779,6 @@ def test_change_speed_cable(duthosts, rand_one_dut_hostname, conn_graph_facts, p
                             pool_oid,
                             pool_size = original_pool_size,
                             shp_size = original_shp_size,
-                            old_speed = original_speed,
-                            new_speed = speed_to_test,
```
Comment (neethajohn) [inline tests/qos/test_buffer.py:738]: why is this removed

Comment (stephenxs) [reply tests/qos/test_buffer.py:738]: As we discussed, we don't need this anymore.
The motivation for passing old/new speed is to adjust the `xon` of the port accordingly in order to calculate the pool size. Now we use the number of lanes which won't be changed during the test.

**Review by neethajohn — APPROVED** [review]

**Review by liat-grozovik — APPROVED** [review]
